api/types.go

```diff
@@ -143,6 +143,12 @@ type Message struct {
 	Thinking  string      `json:"thinking,omitempty"`
 	Images    []ImageData `json:"images,omitempty"`
 	ToolCalls []ToolCall  `json:"tool_calls,omitempty"`
+	ToolResult
```

Comment:
This seems like a great design inline with others:

1. A model can output one or more tool calls
2. For each tool call the API consumer then needs to create a message per tool result, setting these fields

Would it be possible to update the api.md docs?